### ISSUE TITLE
--change-interface doesn't check length of interface name

### ISSUE THREAD

**Issue by spwhitton** [firewalld/firewalld#873]
**What happened**:

The --change-interface option doesn't check that the length of the provided interface name is within the sixteen character limit.  If you use with --permanent or with firewall-offline-cmd then the zone's XML will be updated but then the next --reload will fail with a pile of cryptic nftables JSON.  If you use without --permanent then you get error messages from python-nftables, and nftables gets into an inconsistent state that is difficult to back out of.

**What you expected to happen**:

The command line interface should error out immediately when the provided interface name is greater than sixteen characters, not touching either the runtime or permanent configuration.

**Environment**:
- Firewalld Version: 0.9.3
- Firewalld Backend: nftables
- OS: Debian bullseye

**Comment by kircherlike** [comment]
I think nftables is a program written in C that includes the string termination character as part of the 16-character string.

**Comment by unknowndevQwQ** [comment]
Hi, I'm curious how you make the interface name longer than 15 bytes?

**Comment by spwhitton** [comment]
I can't remember -- is there some kernel limit?

**Comment by unknowndevQwQ** [comment]
> I can't remember -- is there some kernel limit?

Yes, but with altname/alias you can write very long names, but that's an alias.

**Comment by erig0** [comment]
Kernel limit is [IFNAMSIZ](https://www.man7.org/linux/man-pages/man7/netdevice.7.html). I would be great if nftables/iptables supported altname, but I think that's a lot of work.

This bug is a can't fix. It has to be fixed in the kernel (nftables).
